Standardize code across all tutorials

mfl28 opened the pull request:
This resolves prabhuomkar/pytorch-cpp#18. All tutorials now follow a similar code style and format.

prabhuomkar commented on tutorials/basics/pytorch_basics/main.cpp:
```diff
@@ -1,91 +1,244 @@
+
+    // Construct data loader
+    auto dataloader = torch::data::make_data_loader<torch::data::samplers::RandomSampler>(
+        dataset, /*batch size=*/ 64);
```
We can skip the commenting of arguments for all dataloaders

mfl28 replied:
Agreed!

prabhuomkar commented:
Thank you so much @mfl28 but there are lot of changes to track as of now.
Can you send seperate PRs for new advanced tutorials and structuring code overall?

mfl28 commented:
Glad to help! 
I did not add any new advanced tutorials in this PR. It looks like a lot of changes in the diff, but a lot of those are just formatting changes like using 4 (instead of 2)  space indentation in functions, and moving things into a more consistent order in the main file of every tutorial. 
I also moved the neural network class in the intermediate/feedforward_neural_network into its own header (and moved the implementation into an accompanying .cpp) to be consistent with the other tutorials.

Things I actually added:
- Printing the device used (CPU or GPU) for every tutorial (this was missing in all intermediate/advanaced tutorials).
- Some mini-subchapters  from [Python pytorch-basics](https://github.com/yunjey/pytorch-tutorial/blob/master/tutorials/01-basics/pytorch_basics/main.py) (see table of contents) that were missing in the C++ implementation in basics/pytorch-basics/main.cpp:
  - Loading data from numpy (I changed this to "Creating Tensors From Existing Data": This shows how to create tensors from C-style arrays and vectors)
  - Input pipeline
  - Input pipeline for custom dataset (Here I just added a comment that points the reader to the cifar10 dataset implementation in the deep residual network tutorial)
  - Pretrained model
  - Save and load model

I will remove the argument comments later!

prabhuomkar commented:
That's amazing @mfl28 let's work on having small changes every PR so we can dig more into it :) 

Do you have any ideas of making our CI a bit better?
